src/sentry/integrations/github/client.py

```diff
@@ -70,99 +70,99 @@ def get_repo(self, repo: str) -> JSONData:
+                    logger.exception("An unknown error has ocurred.")
 
         return repo_files
 
     def get_trees_for_org(
         self, cache_key: str, gh_org: str, cache_seconds: int = 3600 * 24
     ) -> Dict[str, RepoTree]:
         """
-        This fetches tree representations of all repos for an org.
+        This fetches tree representations of all repos for an org and saves its
+        contents into the cache.
         """
         trees: Dict[str, RepoTree] = {}
         cache_key = f"githubtrees:repositories:{cache_key}:{gh_org}"
-        repo_key = "githubtrees:repo"
-        cached_repositories = cache.get(cache_key, [])
-        if not cached_repositories:
+        repositories = cache.get(cache_key)
+        if not repositories:
             # Simply removing unnecessary fields from the response
             repositories = [
                 {"full_name": repo["full_name"], "default_branch": repo["default_branch"]}
                 for repo in self.get_repositories(fetch_max_pages=True)
             ]
-            # XXX: In order to speed up this function we will need to parallelize this
-            # Use ThreadPoolExecutor; see src/sentry/utils/snuba.py#L358
-            for repo_info in repositories:
-                try:
-                    full_name: str = repo_info["full_name"]
-                    branch = repo_info["default_branch"]
-                    files = self.get_repo_files(full_name, branch)
-                    repo = Repo(full_name, branch)
-                    trees[full_name] = RepoTree(repo, files)
-                    cache.set(f"{repo_key}:{full_name}", trees[full_name], cache_seconds)
-                except Exception:
-                    # Catching the exception ensures that we can make progress with the rest
-                    # of the repositories
-                    logger.exception(f"We have failed to fetch the tree for {repo_info}.")
             cache.set(cache_key, repositories, cache_seconds)
             next_time = datetime.now() + timedelta(seconds=cache_seconds)
             logger.info(f"Caching trees for {gh_org} org until {next_time}.")
-        else:
+
+        # XXX: In order to speed up this function we will need to parallelize this
+        # Use ThreadPoolExecutor; see src/sentry/utils/snuba.py#L358
+        for repo_info in repositories:
+            full_name = repo_info["full_name"]
+            branch = repo_info["default_branch"]
             try:
-                for repo_info in cached_repositories:
-                    repo_tree = cache.get(f"{repo_key}:{repo_info['full_name']}")
-                    # We should not expect that the cache has not been cleared
-                    if not repo_tree:
-                        repo_tree = self.get_repo_files(
-                            repo_info["full_name"],
-                            repo_info["default_branch"],
-                        )
-                    trees[repo_info["full_name"]] = repo_tree
```

Comment:
The logic in here has been simplified with the block below:
<img width="363" alt="image" src="https://user-images.githubusercontent.com/44410/204310050-afce4d95-1d82-47bc-87ff-118ab3ab78ce.png">